src/OSInfo.cpp

```diff
@@ -373,170 +256,137 @@ QString OSInfo::getOSDisplayableVersion() {
+		return QString();
 	}
 
-	// Architecture
-	if (ovi.dwMajorVersion >= 6) {
-		if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_AMD64) {
-			osdispver.append(QLatin1String(" x64"));
-		} else if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_INTEL) {
-			osdispver.append(QLatin1String(" x86"));
-		}
+	return QString::asprintf("%lu.%lu.%lu.%lu", static_cast< unsigned long >(ovi.dwMajorVersion),
+							 static_cast< unsigned long >(ovi.dwMinorVersion),
+							 static_cast< unsigned long >(ovi.dwBuildNumber),
+							 (ovi.wProductType == VER_NT_WORKSTATION) ? 1UL : 0UL);
+#elif defined(Q_OS_MAC)
+	SInt32 major, minor, bugfix;
+	OSErr err = Gestalt(gestaltSystemVersionMajor, &major);
+	if (err == noErr)
+		err = Gestalt(gestaltSystemVersionMinor, &minor);
+	if (err == noErr)
+		err = Gestalt(gestaltSystemVersionBugFix, &bugfix);
+	if (err != noErr)
+		return QString::number(QSysInfo::MacintoshVersion, 16);
+
+	char *buildno = nullptr;
+	char buildno_buf[32];
+	size_t sz_buildno_buf = sizeof(buildno);
+	int ret               = sysctlbyname("kern.osversion", buildno_buf, &sz_buildno_buf, nullptr, 0);
+	if (ret == 0) {
+		buildno = &buildno_buf[0];
 	}
 
-	QString osv;
-#	if QT_VERSION >= 0x050500
-	osv = QString::asprintf(" - %lu.%lu.%lu", static_cast< unsigned long >(ovi.dwMajorVersion),
-							static_cast< unsigned long >(ovi.dwMinorVersion),
-							static_cast< unsigned long >(ovi.dwBuildNumber));
-#	else
-	// sprintf() has been deprecated in Qt 5.5 in favor for the static QString::asprintf()
-	osv.sprintf(" - %lu.%lu.%lu", static_cast< unsigned long >(ovi.dwMajorVersion),
-				static_cast< unsigned long >(ovi.dwMinorVersion), static_cast< unsigned long >(ovi.dwBuildNumber));
-#	endif // QT_VERSION >= 0x050500
-	osdispver.append(osv);
-
-	return osdispver;
+	return QString::asprintf("%lu.%lu.%lu %s", static_cast< unsigned long >(major), static_cast< unsigned long >(minor),
+							 static_cast< unsigned long >(bugfix), buildno ? buildno : "unknown");
 #else
-	return OSInfo::getOSVersion();
+	return QSysInfo::productVersion();
 #endif
 }
 
 void OSInfo::fillXml(QDomDocument &doc, QDomElement &root, const QString &os, const QString &osver,
 					 const QList< QHostAddress > &qlBind) {
-	QDomElement tag;
-	QDomText t;
-	bool bIs64;
-	bool bSSE2 = false;
-	QString cpu_id, cpu_extid;
+	auto tag = doc.createElement(QLatin1String("machash"));
+	root.appendChild(tag);
+	auto t = doc.createTextNode(getMacHash(qlBind));
+	tag.appendChild(t);
 
-	tag = doc.createElement(QLatin1String("machash"));
+	tag = doc.createElement(QLatin1String("arch"));
 	root.appendChild(tag);
-	t = doc.createTextNode(OSInfo::getMacHash(qlBind));
+	t = doc.createTextNode(getArchitecture(true));
 	tag.appendChild(t);
 
 	tag = doc.createElement(QLatin1String("version"));
 	root.appendChild(tag);
-	t = doc.createTextNode(QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+	t = doc.createTextNode(QLatin1String(MUMTEXT(MUMBLE_VERSION_STRING)));
```

Comment:
Does this do what we want ti to do in cases where `MUMBLE_VERSION_STRING` is not defined?

Which btw. seems to always be the case. I can't find a mention of this in the current source code. Do you perhaps mean `MUMBLE_RELEASE_ID`?

Reply:
Sorry, this change is a leftover of the old code.